scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -697,8 +709,11 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
     }
   }
 
-  private def getDeprecatedAnno(v: Version) =
-    q"new scala.deprecated(${Literal(Constant(v.toString))})"
+  private def getDeprecatedAnno(v: Version, why: String = "") =
+    q"new scala.deprecated(${Literal(Constant(v.toString + why))})"
```

Comment:
Slightly beyond the scope of this PR, but why not advertize the "since" version through the argument of the same name? It can be useful for filtering with `-Wconf`

Reply:
see comment above

Reply:
in case i misunderstood it the first time: i can't always provide the "since" parameter.